os/drivers/testcase/test_kthread_stack_protection.c

```diff
@@ -0,0 +1,170 @@
+/****************************************************************************
+ *
+ * Copyright 2020 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <tinyara/testcase_drv.h>
+#include <debug.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <semaphore.h>
+#include <tinyara/kthread.h>
+
+#include "testcase_proto.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+#define	KTHREAD_STACK_NORMAL_SIZE		300
+#define	KTHREAD_STACK_OVERFLOW_SIZE		4096
+
+/****************************************************************************
+ * Private Declarations
+ ****************************************************************************/
+
+static sem_t test_kthread_stack_sem;
+
+/****************************************************************************
+ * Private Function
+ ****************************************************************************/
+
+#if defined(CONFIG_REG_STACK_OVERFLOW_PROTECTION)
+static void kthread_stack_overflow_array(void)
+{
+	volatile char array_normal[KTHREAD_STACK_OVERFLOW_SIZE];
+
+	array_normal[0] = 'a';
+}
+
+static void kthread_stack_prot_inlimit(void)
+{
+	char array_normal[KTHREAD_STACK_NORMAL_SIZE];
+
+	array_normal[0] = 'a';
+}
+
+static void kthread_stack_overflow_func(int *val)
+{
+	char *ptr;
+	int i = 0;
+	volatile char array_normal[KTHREAD_STACK_NORMAL_SIZE];
+
+	ptr = &array_normal[KTHREAD_STACK_NORMAL_SIZE];
+	for (i = 0; i < KTHREAD_STACK_NORMAL_SIZE; i++) {
+		*ptr = 'a';
+		ptr = ptr - 1;
+	}
+
+	*val = *val + 1;
+	if (*val == (KTHREAD_STACK_OVERFLOW_SIZE / KTHREAD_STACK_NORMAL_SIZE)) {
+		return;
+	}
+	/* recursive call to create stack overflow condition */
+	kthread_stack_overflow_func(val);
+}
+#endif
+
+static int kthread_stack_overflow_prot(int argc, FAR char *argv[])
```

Comment:
Same as previous. This is testing stack protection.

Reply:
Have changed name. Please check.